Mathlib/Data/Polynomial/Degree/Lemmas.lean

```diff
@@ -181,6 +181,18 @@ theorem coeff_pow_of_natDegree_le (pn : p.natDegree ≤ n) :
     exact mul_le_mul_of_nonneg_left pn m.zero_le
 #align polynomial.coeff_pow_of_nat_degree_le Polynomial.coeff_pow_of_natDegree_le
 
+theorem coeff_pow_of_natDegree_le_of_eq_ite {o : ℕ}
+    (pn : natDegree p ≤ n) (mno : m * n ≤ o) :
+    coeff (p ^ m) o = if o = m * n then (coeff p n) ^ m else 0 := by
+  split_ifs with h
+  · subst h
+    rw [mul_comm]
```

Comment:
I think this `mul_comm` is a sign that `coeff_pow_of_natDegree_le` should be restated.

Indeed if we restate it as follows:
```lean
theorem coeff_pow_of_natDegree_le (pn : p.natDegree ≤ n) :
    (p ^ m).coeff (m * n) = p.coeff n ^ m := by
  induction' m with m hm
  · simp
  · rw [pow_succ', pow_succ', ← hm, Nat.succ_mul, coeff_mul_of_natDegree_le _ pn]
    refine' natDegree_pow_le.trans (le_trans _ (le_refl _))
    exact mul_le_mul_of_nonneg_left pn m.zero_le
```
then its proof no longer needs to invoke `mul_comm` and the requirement here disappears too.

Reply:
And then this result golfs to:
```lean
theorem coeff_pow_eq_ite_of_natDegree_le_of_le {o : ℕ}
    (pn : natDegree p ≤ n) (mno : m * n ≤ o) :
    coeff (p ^ m) o = if o = m * n then (coeff p n) ^ m else 0 := by
  rcases eq_or_ne o (m * n) with rfl | h
  · simpa only [ite_true] using coeff_pow_of_natDegree_le pn
  · simpa only [h, ite_false] using coeff_eq_zero_of_natDegree_lt $
      lt_of_le_of_lt (natDegree_pow_le_of_le m pn) (lt_of_le_of_ne mno h.symm)
```

Reply:
I accepted this and added a `post-porting note`, since the align statement will no longer point to a theorem with the same signature.